### PR TITLE
cray-libsci/cp2k: add constraints to get consistently linked executables

### PR DESCRIPTION
**Pull request by dev-zero**

### CODE REVIEW

**Comment by dev-zero** [comment]
@lukebroskop maybe you could take a look at the cray-libsci change? For most packages it will only be a small dependency change since many packages check whether or not they have to explicitly link blas/lapack/scalapack libraries and omit that in the presence of Cray compiler wrappers. For CP2K we have to write a rather specific arch file where we explicitly link the math libs.
@shahzebsiddiqui this might change/break the spec tree in e4s once merged

**Review by alalazo — APPROVED** [review]
Waiting for @lukebroskop comment, but LGTM

**Review by lukebroskop — APPROVED** [review]
LGTM!

**Pull request merged**
